### ISSUE TITLE
Creating new tiles in mapping using shift or ctrl creates many new grids and lots of pain.

### ISSUE THREAD

**Issue by IamVelcroboy** [space-wizards/space-station-14#33411]
## Description
A major regression occurred 6-8 months ago making it so `see: title` happens when trying to expand on grids. For a while this was not much of an issue as long as you were building off an existing tile or grid. But it has become increasingly difficult to not create rogue grids while mapping.   

**Reproduction**
use ctrl or shift + tile drag. It's relatively random but can happen with as few as 5 new tiles. 

**Screenshots**
n/a

**Additional context**
n/a

**Comment by IamVelcroboy** [comment]
God, this is getting old. 😩
![image](https://github.com/user-attachments/assets/16534a1b-1f9c-469a-8519-909f58455140)

**Comment by Pixeltheaertist** [comment]
Yeah this is a bit painful. Nothing like spending a long while mapping, saving and initting it and watching in horror as each tile spaces itself because of rogue grids

**Comment by mrhappypenguin** [comment]
I'm having this issue. I want to build a station and shuttles, but I cannot place more than 5x5 at a time without creating multiple grids. Sometimes it occurs with a 2x1 set of plating.

1. Place a single 1x1 plating. 
2. Hold CTRL and place a 5x5 or larger section of plating connected to the previous plating tile.
3. Hold CTRL and place floor tiles of the same size over the new area.
4. Notice some floors cannot be placed due to many newly created grids.
